### ISSUE TITLE
Remove unused code from CATcher

### ISSUE THREAD

**Issue by chunweii** [CATcher-org/WATcher#72]
Since WATcher is a fork from CATcher, there are still many instances of old and unused code from CATcher. Let's remove these codes.

- `DataService` (used to access data.csv in CATcher)
- `src/app/core/services/session-fix-confirmation` (component to ask user for permission to create a new repository)
- `fetchSettingsFile()` and `fetchDataFile()` in `github.service.ts`
- All tests related to the above

There could be more unused code in the project. These are some that I have found

**Comment by vigneshsankariyer1234567890** [comment]
Adding some more after my personal investigation
- All files in `src/app/core/models/conflict`
- All files in `src/app/core/models/generators`
- Seems like a lot of models in `src/app/core/models/github` are used by unused methods in `issue.service.ts`
- and many more...

Perhaps we can break up the task into sub-tasks by folders? That way the tasks are much more easier to complete, rather than having to go through all the folders within one issue. Wdyt @chunweii ?

**Comment by chunweii** [comment]
Yes, we should break fix this issue in several PRs. Removing all unused code in 1 PR might be too big of a change.

**Comment by chia-yh** [comment]
I noticed `profileForm` in `session-selection.component.ts` doesn't appear to be used currently, are there any plans on utilising `Profile`s in WATcher in the future? There are also some methods associated with it (e.g. `initProfileForm`, `onProfileSelect`) that could be candidates for removal in the case that there are no future plans for using `Profile` with WATcher.
